tests/unit/cloudformation/test_connection.py

```diff
@@ -149,6 +150,11 @@ def test_update_stack_all_args(self):
             'NotificationARNs.member.2': 'arn:notify2',
             'Parameters.member.1.ParameterKey': 'KeyName',
             'Parameters.member.1.ParameterValue': 'myKeyName',
+            'Parameters.member.2.ParameterKey': 'KeyName2',
+            'Parameters.member.2.ParameterValue': '',
```

Comment:
I'll try and find some time to remove and test the above change. If I can update the logic I will. Otherwise we might have to leave it as is.